### PR TITLE
Introduce maxAwardedTotal achievement rule option

### PR DESCRIPTION
**Pull request by cmfcmf**
This splits the `maxAwarded` option into two: `maxAwarded` and `maxAwardedTotal`, like discussed in #27. `maxAwarded` works as before, but `maxAwardedTotal` now looks at the total amount of the particular achievement ever granted to the user. This is achieved (hehe) by splitting the `amount` database field in the `achievements` table into `current_amount` and `total_amount`. `current_amount` is set to `0` when the achievement is replaced by another achievement, but `total_amount` only counts upwards.
This makes `replaces` a bit more useful.

### CODE REVIEW

File: test/hooks/achievement-rule-checker.test.js [hunk 1]
```diff
@@ -144,11 +150,12 @@ describe('\'achievement-rule-checker\' hook', () => {
...
     });
 
-    assert.deepEqual(result, []);
+    assert.lengthOf(result, 1);
+    assert.deepInclude(result[0], {name: achievement_name, current_amount: 0, total_amount: 2});
   });
 
 
-  it('gives achievement maxAwarded times', async () => {
+  it('gives achievement maxAwardedTotal times', async () => {
```
Comment (frederike-ramin) [inline test/hooks/achievement-rule-checker.test.js]: Now there is no test for `maxAwarded` anymore although we should probably test that as well. Maybe even combined with `maxAwardedTotal`.

Comment (cmfcmf) [reply test/hooks/achievement-rule-checker.test.js]: You're right, I was a bit lazy there. I've now tested both `maxAwarded` and `maxAwardedTotal` using a `for`-loop.

File: src/achievement-rule.js [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+      maxAwarded = maxAwardedTotal;
+    } else if (maxAwarded !== undefined && maxAwardedTotal === undefined) {
+      // In case only maxAwarded is set, we need to set maxAwardedTotal to +Infinity.
+      maxAwardedTotal = Number.POSITIVE_INFINITY;
+    }
+    if (maxAwarded > maxAwardedTotal) {
+      throw new Error(`The achievement "${this.name}" has maxAwarded > maxAwardedTotal (${this.maxAwarded} > ${this.maxAwardedTotal}).`);
+    }
+    this.maxAwarded = maxAwarded;
+    this.maxAwardedTotal = maxAwardedTotal;
```
Comment (cmfcmf) [inline src/achievement-rule.js]: I moved these classes into a separate file to make the `AchievementRule` easier to test. Apart from the `maxAwarded<Total>` logic above and around line 50, I didn't change anything.

**Comment by cmfcmf** [comment]
@frederike-ramin could you please review again? We also need another reviewer, maybe @KBorchar?

**Comment by cmfcmf** [comment]
This is ready for another review.

**Pull request merged**